Решил задачу

vzeten opened the pull request:
Потренируем StreamAPI 1

Commit:
Решил задачу
Потренируем StreamAPI 1

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    //Найти среднее значение чисел в списке.
+    public static double countAverage(List<Integer> list) {
```
префикс count наводит на мысль о том, что мы получим количество
Тут лучше find, get, calc, calculate и подобное

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    //Отсортировать список строк по длине.
+    public static List<String> sortStringByLong(List<String> list) {
```
тут не byLong, а byLength

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    //Отфильтровать список строк и оставить только те, которые содержат определённую подстроку.
+    public static List<String> filterBySubstring(List<String> list, String substring) {
```
лучше уточнить, как именно проходит фильтрация, что именно filterStringContainsSubstring
Ведь ещё может быть много вариантов разных - startsWith, endsWith и т.д.

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+                .filter(str -> !str.isEmpty())
+                .filter(str ->
+                        Character.toLowerCase(str.charAt(0))
+                                == Character.toLowerCase(ch))
```
Тут проще использовать str.startsWith
К тому же, ещё вопрос - надо ли игнорировать регистр ? это тоже можно отразить в имени метода суффиксами ignoreCase, caseSensitive

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+                .filter(Objects::nonNull)
+                .mapToInt(num -> num)
+                .max()
+                .orElse(0);
```
Тут возвращать 0 - некорректно, ведь это может быть воспринято, как реальное максимальное значение
Лучше выбрасывать исключение orElseThrow или возвращать optional

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+                .filter(Objects::nonNull)
+                .mapToInt(num -> num)
+                .average()
+                .orElse(0.0);
```
Тут аналогично с возвращением нуля

iBerbasov commented on src/main/java/school/faang/bjs2_88163_streamAPI_1/StreamMethods.java:
```diff
@@ -0,0 +1,89 @@
+                .filter(Objects::nonNull)
+                .mapToInt(n -> n)
+                .filter(n -> n > num)
+                .min();
```
А вот тут Optional - отлично подходит

Commit:
Решил задачу
Потренируем StreamAPI 1

Commit:
Добавил .orElseThrow(() вместо .orElse();
заменил название метода с countAverage на findAverage;
убрал нечувствтельность к регистру в countStringsStartingWith;
переименовал метод countStringsStartingWith на countStringsStartingWithCaseSensitive;
переименовал метод filterBySubstring на filterStringContainsSubstringIgnoreCase;
переименовал метод sortStringByLong на sortStringByLenght;
